src/Dfe.PlanTech.Web/Authorisation/ApiKeyAuthorisationFilter.cs

```diff
@@ -0,0 +1,26 @@
+using Dfe.PlanTech.Domain.Persistence.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Dfe.PlanTech.Web.Authorisation;
+
+public class ApiKeyAuthorisationFilter([FromServices] CacheRefreshConfiguration cacheRefreshConfiguration) : IAuthorizationFilter
+{
+    private const string ApiKeyName = "X-WEBSITE-CACHE-CLEAR-API-KEY";
+
+    public void OnAuthorization(AuthorizationFilterContext context)
+    {
+        if (!context.HttpContext.Request.Headers.TryGetValue(ApiKeyName, out var providedApiKey))
+        {
+            context.Result = new UnauthorizedResult();
+            return;
+        }
+
+        var validApiKey = cacheRefreshConfiguration.ApiKey;
+        if (validApiKey is null || !validApiKey.Equals(providedApiKey))
```

Comment:
What if `cacheRefreshConfiguration.ApiKey` and `X-WEBSITE-CACHE-CLEAR-API-KEY` are both empty strings?

Reply:
ive made this a case where it'll be unauthorised